perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -356,7 +362,7 @@ public AttributesManagerBlImpl(AttributesManagerImplApi attributesManagerImpl) {
 	}
 
 	@Override
-	public List<Attribute> getAttributes(PerunSession sess, Group group, Member member, Resource resource, List<String> attrNames, boolean workWithUserAttributes) throws InternalErrorException, MemberResourceMismatchException, GroupResourceMismatchException {
+	public List<Attribute> getAttributes(PerunSession sess, Group group, Member member, Resource resource, List<String> attrNames, boolean workWithUserAttributes) throws InternalErrorException, MemberResourceMismatchException, GroupResourceMismatchException, MemberGroupMismatchException {
```

Comment:
Why are you adding exception MemberGroupMismatchException to the methods, where is no control of it?

Reply:
Its enforced by the calling ```getAttributes(sess, member, group)``` within the method, but you are right that we should check it, since for the case of non-empty ```attrNames``` it wouldn't be checked at all.

Reply:
Fixed.

Reply:
It is not just in this method. So it was more a rhetorical question. If they need to have this check just because of under-layer method, then it is ok.

Reply:
Ok. I'll check other methods, to be sure check is not skipped like it was there.